src/Silex/Provider/Psr11ServiceProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Silex\Provider;
+
+use Pimple\Container;
+use Pimple\Psr11\Container as PsrContainer;
+use Pimple\ServiceProviderInterface;
+use Silex\Provider\Psr11\ContainerValueResolver;
+use Silex\Provider\Psr11\ControllerResolver;
+use Symfony\Component\HttpKernel\Kernel;
+
+/**
+ * Provides a PSR-11 container and argument value resolver.
+ *
+ * @author Pascal Luna <skalpa@zetareticuli.org>
+ */
+class Psr11ServiceProvider implements ServiceProviderInterface
+{
+    public function register(Container $app)
+    {
+        $app['psr11'] = function ($app) {
```

Comment:
psr11 is a very unclear name for beginners and people ignoring php-fig.

"container" or "service_locator" would be more explicit on the purpose of this service.